File: Java_Completo_Programacao_Orientada_a_Objetos_Nelio Alves/Secao_9_Construtores_this_sobrecarga_encapsulamento/Exercicio_de_fixacao_secao9/src/entities/Account.java
package entities;

public class Account {

	private int accNumber;
	private String holder;
	private double balance;
	
	public Account() {
	}
	
	public Account(int accNumber, String holder) {
		this.accNumber = accNumber;
		this.holder = holder;
	}
	
	public Account(int accNumber, String holder, double initialDeposit) {
		this.accNumber = accNumber;
		this.holder = holder;
		deposit(initialDeposit);
	}
	
	public int getAccNumber() {
		return accNumber;
	}
	
	public String getHolder() {
		return holder;
	}
	
	public void setHolder(String holder) {
		this.holder = holder;
	}
	
	public double getBalance() {
		return balance;
	}
	
	public void deposit(double amount) {
		this.balance += amount;
	}
	
	public void withdraw(double amount) {
		this.balance -= amount + 5.0;
	}
	
	public String toString() {
		return "Account " 
	           + accNumber 
	           + ", Holder: " 
	           + holder 
	           + ", Balance: $ " 
	           + String.format("%.2f", balance);
	}
	
}

File: C_Sharp_completo_programacao_orientada_a_objetos/Secao-10-Aula-138-Classes-e-metodos-selados/Secao-10-Aula-138-Classes-e-metodos-selados/Entities/Account.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Secao_10_Aula_138_Classes_e_metodos_selados.Entities
{
    internal class Account
    {
        //Encapsulando com o private.
        public int Number { get; private set; }
        public string Holder { get; private set; }
        //protected, o set só poder ser usado nessa classe Account ou em sua subclasse, no caso BusinessAccount
        public double Balance { get; protected set; }

        public Account()
        {
        }
        public Account(int number, string holder, double balance)
        {
            Number = number;
            Holder = holder;
            Balance = balance;
        }

        //virtual, significa que esse método pode ser sobreposto nas subclasses.
        //a regra desse método é que amount vai receber + 5.0
        public virtual void Withdraw(double amount)
        {
            Balance -= amount + 5.0;
        }

        public void Deposit(double amount)
        {
            Balance += amount;
        }
    }
}